lib/filesystem.js

```diff
@@ -73,22 +73,35 @@ class Filesystem {
       return callback()
     }
 
-    const tr = options.transform && options.transform(p)
-    if (tr) {
-      return tmp.file(function (err, path) {
-        if (err) { return handler() }
-        const out = fs.createWriteStream(path)
-        const stream = fs.createReadStream(p)
-
-        stream.pipe(tr).pipe(out)
-        return tr.on('end', function () {
-          file.transformed = {
-            path,
-            stat: fs.lstatSync(path)
-          }
+    if (options.transform instanceof Function) {
+      options.transform = [options.transform]
+    }
+
+    if (options.transform instanceof Array && options.transform.length) {
+      options.transform = options.transform.filter(() => true)
```

Comment:
@benlancaster Neat trick, would consider it a bit overly defensive though.  Passing arrays like that around would blow up most code in NPM 😆